woocommerce/woocommerce-android
# Issue/11508 coupons card 2

Part of: #11508 


### Description
This PR adds the following to the coupons Card:
- Loading UI.
- Empty View.
- Coupons List UI.
- Navigation to the coupons list.
- Navigation to the coupon details when a coupon is clicked.

### Testing instructions
1. Prepare some test data: create some coupons, and add them to some orders (you can do it from wp-admin even for old orders).
2. Open the app.
3. Switch to different date ranges, and confirm the coupons are loaded, and that the empty view is shown when needed.
4. Click on "View all coupons"
5. Confirm the coupons list is shown.
6. Go back, and click on one of the coupons.
7. Confirm the coupon details screen is shown.


### Images/gif
[Screen_recording_20240519_123512.webm](https://github.com/woocommerce/woocommerce-android/assets/1657201/12fae0bc-d4ae-4791-ba2c-133cb4e99f68)
